src/Controller/SearchResultsController.php

```diff
@@ -73,33 +84,83 @@ public function SearchForm(): Form
 
     public function SearchResults(): ?Results
     {
+        // Local cache to make sure we don't perform this task more than one (@see QuerySuggestions())
+        if ($this->results) {
+            return $this->results;
+        }
+
         $request = $this->getRequest();
-        // Field names (as configured)
+        // Keyword (aka search query string) field (as configured)
         $fieldKeywords = $this->config()->get('field_keywords');
-        $fieldPagination = $this->config()->get('field_pagination');
-        // The index variant that we are fetching records from (as defined under `indexes` in search.yml)
-        $index = $this->config()->get('index_variant');
-        // How many records we want to display per page
-        $perPage = $this->config()->get('per_page');
         // The keywords that we want to search
         $keywords = $request->getVar($fieldKeywords);
-        // Pagination (if supplied)
-        $start = $request->getVar($fieldPagination) ?? 0;
 
         // No results unless we search for something
         if (!$keywords) {
             return null;
         }
 
-        $service = SearchService::create();
+        // Pagination field (as configured)
+        $fieldPagination = $this->config()->get('field_pagination');
+        // The index variant that we are fetching records from (as defined under `indexes` in search.yml)
+        $index = $this->config()->get('index_variant');
+        // How many records we want to display per page
+        $perPage = $this->config()->get('per_page');
+        // Pagination (if supplied)
+        $start = $request->getVar($fieldPagination) ?? 0;
+
+        // Instantiate our service
+        $service = SearchService::singleton();
```

Comment:
This logic is all the same, it's just that I wanted to move it below the `if (!$keywords)` condition so that we exit slightly earlier.